pkg/logs/input/docker/launcher.go

```diff
@@ -158,15 +159,39 @@ func (l *Launcher) run() {
 			containerID := service.Identifier
 			l.stopTailer(containerID)
 			delete(l.pendingContainers, containerID)
-		case containerId := <-l.erroredContainerID:
-			go l.restartTailer(containerId)
+		case containerID := <-l.erroredContainerID:
+			go l.restartTailer(containerID)
 		case <-l.stop:
 			// no docker container should be tailed anymore
 			return
 		}
 	}
 }
 
+// overrideSource create a new source with
+func (l *Launcher) overrideSource(container *Container, source *config.LogSource) *config.LogSource {
+	if source.Name != config.ContainerCollectAll {
```

Comment:
this is not really safe, we should probably compare the pointers ?

Reply:
and move https://github.com/DataDog/datadog-agent/pull/2704/files#diff-2325533c5d014118fdb0b16567c0b316R178 at the very top of the function ?

Reply:
+ I think those checks should be used in the caller

Reply:
It's not unsafe because you can't have a `nil` source here. 

`overrideSource` is called by `startTailer` that cannot have an `nil` source:
Either https://github.com/DataDog/datadog-agent/blob/3365c74c868bd74071f6ed29fac88e8a1261c2f1/pkg/logs/input/docker/launcher.go#L123-L125
or 
https://github.com/DataDog/datadog-agent/blob/3365c74c868bd74071f6ed29fac88e8a1261c2f1/pkg/logs/input/docker/launcher.go#L132-L140 and the scheduler won't push a `nil` source

Reply:
>and move DataDog/datadog-agent/pull/2704/files#diff-2325533c5d014118fdb0b16567c0b316R178 at the very top of the function ?

I'd rather have this like this if we do the check in the calling function `startTailer` because we still need a reference to the old source in the `startTailer` method